walk-in-interview/src/main/webapp/update-job/script.js

```diff
@@ -0,0 +1,438 @@
+
+  const submitButton = document.getElementById('update');
+  submitButton.setAttribute('value', UPDATE_JOB_STRINGS['update']);
+  submitButton.setAttribute('type', 'submit');
+
+  const job = getJobFromId(jobId);
+
+  const jobTitle = document.getElementById('title');
+  const jobTitleContent = job.jobTitle;
+  jobTitle.setAttribute('type', 'text');
+  jobTitle.setAttribute('value', jobTitleContent);
+
+  const jobDescription = document.getElementById('description');
+  const jobDescriptionContent = job.jobDescription;
+  jobDescription.setAttribute('type', 'text');
+  jobDescription.innerText = jobDescriptionContent;
+
+  const jobAddress = document.getElementById('address');
+  const jobAddressContent = job.jobLocation.address;
+  jobAddress.setAttribute('type', 'text');
+  jobAddress.setAttribute('value', jobAddressContent);
+
+  const postalCode = document.getElementById('postal-code');
+  const postalCodeContent = job.jobLocation.postalCode;
+  postalCode.setAttribute('type', 'text');
+  postalCode.setAttribute('value', postalCodeContent);
+    
+  const requirementsTitle = document.getElementById('requirements-title');
+  requirementsTitle.innerText = STRINGS['requirements-title'];
+  const requirements = job.requirements;
+  renderRequirementsList(requirements);
+
+  const payTitle = document.getElementById('pay-title');
+  payTitle.innerText = STRINGS['pay-title'];
+  const jobPayFrequency = job.jobPay.paymentFrequency;
+  renderJobPayFrequencyOptions(jobPayFrequency);
+    
+  const jobPayMin = document.getElementById('pay-min');
+  const jobPayMinContent = job.jobPay.min;
+  jobPayMin.setAttribute('type', 'number');
+  jobPayMin.setAttribute('value', jobPayMinContent);
+
+  const jobPayMax = document.getElementById('pay-max');
+  const jobPayMaxContent = job.jobPay.max;
+  jobPayMax.setAttribute('type', 'number');
+  jobPayMax.setAttribute('value', jobPayMaxContent);
+
+  const durationTitle = document.getElementById('duration-title');
+  durationTitle.innerText = STRINGS['duration-title'];
+  const jobDuration = job.jobDuration;
+  renderJobDurationOptions(jobDuration);
+
+  const expiryTitle = document.getElementById('expiry-title');
+  expiryTitle.innerText = STRINGS['expiry-title'];
+  const jobExpiry = document.getElementById('expiry');
+  const jobExpiryTimestamp = job.postExpiryTimestamp;
+  jobExpiry.setAttribute('type', 'date');
+  renderJobExpiryLimits(jobExpiryTimestamp);
+
+  // Resets the error to make sure no error msg initially present.
+  setErrorMessage(/* errorMessageElementId= */'error-message', /* msg= */'', /* includesDefault= */false);
+}
+
+/**
+ * Gets the Job post given its id.
+ * 
+ * @param {String} jobId Cloud Firestore Job Id.
+ * @returns Job json.
+ */
+function getJobFromId(jobId) {
+  // Only run it for selenium test.
+  // return getJobForSeleniumTest();
+
+  // TODO(issue/53): run the web page to test once doGet finishes in JobServlet
+  const url = `/jobs?jobId=${jobId}`;
```

Comment:
Should we have some constant file for all API endpoints listed in the API design doc?